### PR TITLE
[SelectiveMasking] install masks for QgsTemplatedLineSymbolLayer

### LINKED ISSUE #57564: Marker Line masking not working for layout prints or PDF exports
### What is the bug or the crash?

Marker Lines can be set to be masked and this works on the screen but doesn't work for prints or exports to vector or image formats.  Simple Lines can be masked correctly.

### Steps to reproduce the issue

View the test project at [label mask test.zip](https://github.com/qgis/QGIS/files/15441719/label.mask.test.zip) or:
1. Create a new project
2. Create Layer -> New Shapefile Layer -> Geometry type -> LineString
3. Edit the LineString to add a single line with a few vertices
4. Properties -> Symbology to Single Symbol with both a Marker Line and a Simple Line visible
5. Vector -> Geometry Tools -> Extract Vertices to create a new layer with just the points of the LineString
6. Points layer -> Properties-> Labels -> Placement -> Offset from Point 0mm, Quadrant Centre
7. Points layer -> Properties-> Mask -> Enable Mask, 2mm
8. LineString layer -> Properties ->Masks: source to Label mask and check both the Simple Line and Marker Line to be Masked.
9. Observe that masking is working on-screen
10. Project -> Export to PDF and see that the masking fails for the markers but works for the simple line.

### Versions

<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.0//EN" "http://www.w3.org/TR/REC-html40/strict.dtd">
<html><head><meta http-equiv="Content-Type" content="text/html; charset=utf-8" /><style type="text/css">
p, li { white-space: pre-wrap; }
</style></head><body>

QGIS version | 3.36.3-Maidenhead | QGIS code revision | 2df96554
-- | -- | -- | --
Qt version | 5.15.13
Python version | 3.12.3
GDAL/OGR version | 3.9.0
PROJ version | 9.4.0
EPSG Registry database version | v11.004 (2024-02-24)
GEOS version | 3.12.1-CAPI-1.18.1
SQLite version | 3.45.1
PDAL version | 2.6.3
PostgreSQL client version | 16.2
SpatiaLite version | 5.1.0
QWT version | 6.2.0
QScintilla2 version | 2.14.1
OS version | Windows 10 Version 2009
  |   |   |  
Active Python plugins
db_manager | 0.1.20
grassprovider | 2.12.99
MetaSearch | 0.3.6
processing | 2.12.99

</body></html>QGIS version
3.36.3-Maidenhead
QGIS code revision
[2df96554](https://github.com/qgis/QGIS/commit/2df96554)
Qt version
5.15.13
Python version
3.12.3
GDAL/OGR version
3.9.0
PROJ version
9.4.0
EPSG Registry database version
v11.004 (2024-02-24)
GEOS version
3.12.1-CAPI-1.18.1
SQLite version
3.45.1
PDAL version
2.6.3
PostgreSQL client version
16.2
SpatiaLite version
5.1.0
QWT version
6.2.0
QScintilla2 version
2.14.1
OS version
Windows 10 Version 2009

Active Python plugins
db_manager
0.1.20
grassprovider
2.12.99
MetaSearch
0.3.6
processing
2.12.99

### Supported QGIS version

- [X] I'm running a supported QGIS version according to [the roadmap](https://www.qgis.org/en/site/getinvolved/development/roadmap.html#release-schedule).

### New profile

- [X] I tried with a new [QGIS profile](https://docs.qgis.org/latest/en/docs/user_manual/introduction/qgis_configuration.html#working-with-user-profiles)

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by troopa81**
Fixes #57564

### CODE REVIEW

**Review by nyalldawson — APPROVED** [review]

**Comment by nyalldawson** [comment]
@troopa81 to fix the test/segfault here i'd suggest using the SafePainter class from https://github.com/qgis/QGIS/blob/master/tests/src/python/test_qgsgeometrypaintdevice.py#L45 instead of raw painters throughout the selective masking test class. That should allow the test to exit gracefully and write out the full test report instead of just crashing when a failure occurs.

**Comment by troopa81** [comment]
unrelated test failure, segmentation faut has been fixed. 

@nyalldawson Do you mind merging?

File: tests/src/python/test_selective_masking.py [hunk 1]
```diff
@@ -1358,7 +1358,9 @@ def test_markerline_masked(self):
         circle_symbol = QgsMarkerSymbol.createSimple({'size': '3'})
         sl.setSubSymbol(circle_symbol)
 
-        self.lines_layer.setRenderer(QgsSingleSymbolRenderer(QgsLineSymbol([sl])))
```
Comment (nyalldawson) [inline tests/src/python/test_selective_masking.py]: @troopa81 FYI, this was apparently fixed in newer sip versions, so eventually this approach will work 😝

Comment (troopa81) [reply tests/src/python/test_selective_masking.py]: So that explain why it works on my computer (Debian SIP 6.8.3) and not in CI (SIP 4.19.25)

**Pull request merged**